bootstrap/main.py

```diff
@@ -0,0 +1,56 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tempfile
+
+import argparse
+import pathlib
+
+from cli import main_menu, do_one_click_build
+from minifi_option import parse_minifi_options
+from package_manager import get_package_manager
+
+if __name__ == '__main__':
+    with tempfile.TemporaryDirectory() as cmake_cache_dir:
+        parser = argparse.ArgumentParser()
+        parser.add_argument('--noconfirm', action="store_true", default=False,
+                            help="Bypass any and all “Are you sure?” messages.")
+        parser.add_argument('--minifi_options', default="", help="Overrides the default minifi options during the "
+                                                                 "initial parsing")
+        parser.add_argument('--cmake_options', default="", help="Appends this to the final cmake command")
+        parser.add_argument('--skip_compiler_install', action="store_true", default=False,
+                            help="Skips the installation of the default compiler")
```

Comment:
Long options usually use dashes as word separator. What do minifi options get passed to?

Reply:
We pass that directly to the first cmake command that parses the minifioptions cmake file

Reply:
I don't really understand the answer, so I'll try to rephrase my question: We should use `--minifi-options` instead of `--minifi_options`, `--cmake-options` instead of `--cmake_options`, and `--skip-compiler-install` instead of `--skip_compiler_install`, because in command line long options, dashes are used as the word separator in most cases.
I'm not sure how cmake fits here? These long option names are parsed by python, aren't they?

Reply:
sry misunderstood the question, I see your point I changed that in https://github.com/apache/nifi-minifi-cpp/pull/1681/commits/faabed2da474a73ae6f3761eeccf128cd09426e3